src/settings/TemplateDetailsViewer.ts

```diff
@@ -17,64 +15,66 @@ import { EntityClient } from "../api/common/EntityClient"
 import { TEMPLATE_SHORTCUT_PREFIX } from "../templates/model/TemplatePopupModel"
 import type { UpdatableSettingsDetailsViewer } from "./SettingsView"
 import { ActionBar } from "../gui/base/ActionBar.js"
+import { htmlSanitizer } from "../misc/HtmlSanitizer.js"
 
 export class TemplateDetailsViewer implements UpdatableSettingsDetailsViewer {
-	isReadOnly: lazy<boolean>
-	readonly renderView: UpdatableSettingsDetailsViewer["renderView"]
+	private readonly sanitizedContents: Array<{ text: string; language: string }>
 
-	constructor(template: EmailTemplate, entityClient: EntityClient, isReadOnly: lazy<boolean>) {
-		this.isReadOnly = isReadOnly
-		this.renderView = () => {
-			return m("#user-viewer.fill-absolute.scroll.plr-l.pb-floating", [
-				m(".flex.mt-l.center-vertically", [
-					m(".h4.text-ellipsis", template.title),
-					!this.isReadOnly()
-						? m(ActionBar, {
-								buttons: [
-									{
-										title: "edit_action",
-										icon: Icons.Edit,
-										click: () => this.editTemplate(template),
-									},
-									{
-										title: "remove_action",
-										icon: Icons.Trash,
-										click: () => this.deleteTemplate(entityClient, template),
-									},
-								],
-						  })
-						: null,
-				]),
-				m("", [
-					m(TextField, {
-						label: "shortcut_label",
-						value: TEMPLATE_SHORTCUT_PREFIX + neverNull(template.tag),
-						disabled: true,
-					}),
-					template.contents.map((emailTemplateContent) => {
-						const language = languageByCode[getLanguageCode(emailTemplateContent)]
-						return m(".flex.flex-column", [
-							m(".h4.mt-l", lang.get(language.textId)),
-							m(".editor-border.text-break", m.trust(emailTemplateContent.text)),
-						])
-					}),
-				]),
-			])
-		}
+	constructor(private readonly template: EmailTemplate, private readonly entityClient: EntityClient, readonly isReadOnly: lazy<boolean>) {
+		this.sanitizedContents = template.contents.map((emailTemplateContent) => ({
+			text: htmlSanitizer.sanitizeHTML(emailTemplateContent.text, { blockExternalContent: false, allowRelativeLinks: true }).html,
+			language: lang.get(languageByCode[getLanguageCode(emailTemplateContent)].textId),
```

Comment:
is storing the translated content here a problem? the view should never survive across the lang switch?

Reply:
this is a viewer that is constructed manually so it might actually survive?

Reply:
that's right, I amended the commit and rebased